### PR TITLE
Add support for vDPA devices management

### PR DESCRIPTION
**Pull request by ykulazhenkov**
[vDPA](https://vdpa-dev.gitlab.io/) - virtio Data Path Acceleration

Following functions are supported by the implementation:

**VDPANewDev** - adds new VDPA device
Equivalent to: `vdpa dev add name <name> mgmtdev <mgmtBus>/<mgmtName> [params]`

**VDPADelDev** removes VDPA device
Equivalent to: `vdpa dev del <name>`

**VDPAGetDevList** returns list of VDPA devices
Equivalent to: `vdpa dev show`

**VDPAGetDevByName** returns VDPA device selected by name
Equivalent to: `vdpa dev show <name>`

**VDPAGetDevConfigList** returns list of VDPA devices configurations
Equivalent to: `vdpa dev config show`

**VDPAGetDevConfigByName** returns VDPA device configuration selected by name
Equivalent to: `vdpa dev config show <name>`

**VDPAGetDevVStats** returns vstats for VDPA device
Equivalent to: `vdpa dev vstats show <name> qidx <queueIndex>`

**VDPAGetMGMTDevList** returns list of mgmt devices
Equivalent to: `vdpa mgmtdev show`

**VDPAGetMGMTDevByBusAndName** returns mgmt devices selected by bus and name
Equivalent to: `vdpa mgmtdev show <bus>/<name>`

### CODE REVIEW

**Review by adrianchiris — APPROVED** [review]
looks good !

@amorenoz this work can simplify [govdpa ](https://github.com/k8snetworkplumbingwg/govdpa/tree/master/pkg/kvdpa) lib

could you take a look ? as this work is a superset of what was done for kernel vdpa in k8snetworkplumbingwg/govdpa

File: vdpa_linux.go [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+// dump all devices if dev is nil
+func (h *Handle) vdpaMGMTDevGet(bus, dev *string) ([]*VDPAMGMTDev, error) {
+	var extraFlags int
+	var attrs []*nl.RtAttr
+	if dev != nil {
+		attrs = append(attrs,
+			nl.NewRtAttr(nl.VDPA_ATTR_MGMTDEV_DEV_NAME, nl.ZeroTerminated(*dev)),
+		)
+		if bus != nil {
```
[the remark this answers is not in the document]
Comment (ykulazhenkov) [reply vdpa_linux.go:331]: yes, example is vdpa_sim device. It doesn't have a bus

File: vdpa_linux.go [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+
+// VDPANewDev adds new VDPA device
+// Equivalent to: `vdpa dev add name <name> mgmtdev <mgmtBus>/mgmtName [params]`
+func (h *Handle) VDPANewDev(name, mgmtBus, mgmtName string, params VDPANewDevParams) error {
+	attrs := []*nl.RtAttr{
+		nl.NewRtAttr(nl.VDPA_ATTR_DEV_NAME, nl.ZeroTerminated(name)),
+		nl.NewRtAttr(nl.VDPA_ATTR_MGMTDEV_DEV_NAME, nl.ZeroTerminated(mgmtName)),
+	}
+	if mgmtBus != "" {
+		attrs = append(attrs, nl.NewRtAttr(nl.VDPA_ATTR_MGMTDEV_BUS_NAME, nl.ZeroTerminated(mgmtBus)))
```
[the remark this answers is not in the document]
Comment (ykulazhenkov) [reply vdpa_linux.go:360]: yes, for some devices it can be not set

[the remark this answers is not in the document]
Comment (adrianchiris) [reply vdpa_linux.go:360]: ack, thx.

**Comment by aboch** [comment]
Unless each commit brings in a independent feature/fix, please have one commit per PR. We have been following this logic in this repo.

**Comment by ykulazhenkov** [comment]
> Unless each commit brings in a independent feature/fix, please have one commit per PR. We have been following this logic in this repo.

Thank you @aboch,
I squashed all commits to single one

**Comment by adrianchiris** [comment]
@aboch @vishvananda could you take a look at this PR ? :)

**Comment by aboch** [comment]
LGTM

**Pull request merged**